pkg/sdn/plugin/proxy.go

```diff
@@ -98,7 +103,39 @@ func (proxy *OsdnProxy) watchEgressNetworkPolicies() {
 	})
 }
 
+func (proxy *OsdnProxy) watchNetNamespaces() {
+	RunEventQueue(proxy.osClient, NetNamespaces, func(delta cache.Delta) error {
+		netns := delta.Object.(*osapi.NetNamespace)
+		name := netns.ObjectMeta.Name
+
+		glog.V(5).Infof("Watch %s event for NetNamespace %q", delta.Type, name)
+		proxy.idLock.Lock()
+		defer proxy.idLock.Unlock()
+		switch delta.Type {
+		case cache.Sync, cache.Added, cache.Updated:
+			proxy.ids[name] = netns.NetID
+		case cache.Deleted:
+			delete(proxy.ids, name)
+		}
+		return nil
+	})
+}
 func (proxy *OsdnProxy) updateEgressNetworkPolicy(policy osapi.EgressNetworkPolicy) {
+	ns := policy.Namespace
+	ignorePolicy := false
+	func() {
+		proxy.idLock.Lock()
+		defer proxy.idLock.Unlock()
```

Comment:
I would have a slight preference for splitting this out into a real helper function rather than an inline func() but if you really prefer it this way you can keep it